### DISCUSSION TITLE
Child process

### DISCUSSION THREAD

**Discussion by trimechee** [Floorp-Projects/Floorp#543] [Feature Request]
Hello, i hope to use this extension to open media player to play youtube videos and video from other sites :

https://github.com/andy-portmen/external-application-button/

the problem with firefox based browser, we can open video of the browser wih media player but there is child process in the background and when we close firefox, the media player closes , we hope our beloved Floorp browser will fix this big problem, chromium browsers doesnt have this problem, thank you !

https://webextension.org/listing/external-application-button.html

Maybe it can help : 

""As I have already stated, there is no fix for this (or at least I am not aware of a way to fix this). Firefox just opens all the executables as its own child process and when the parent process is getting closed, all the sub-processes are being closed. You can still play around with the NodeJS native client which is basically very simple JavaScript code and see if you can find a way to prevent this to happen. "

"[Firefox Only] Can you explain why all the applications executed from this extension in Firefox get closed when the browser is closed? Is there a way to prevent this from happening?

This behavior relates to how Firefox handles native applications as child processes of Firefox. When the parent process is closed, the operating system (Windows only) kills all the child processes, including those launched by this extension. There is no easy way to bypass this restriction. However, a workaround is to use a third-party program called [RunFromProcess](https://www.nirsoft.net/utils/run_from_process.html) by NirSoft that can change the parent process of a child process. You can call this program instead of your native one and ask it to assign another parent process for the application. By doing this, when Firefox is closed, the os does not kill the child process because its parent is still running. You can set a persistent process such as "explorer.exe" as the parent to prevent it from being closed later."

**Comment by creeper-0910** [comment]
To accommodate this, there are more disadvantages such as difficulty of maintenance than advantages, so it will not be implemented.

**Comment by trimechee** [comment]
I understand, thank you

**Comment by trimechee** [comment]
@creeper-0910 I talked to another developers of Firefox forks, we hope if they succeed to incorporate this feature , we hope Floorp browser will incorporate it, i use 

https://ys.3dyd.com/home/

and 

https://github.com/vzhd1701/gridplayer

and 


https://github.com/zaps166/QMPlay2

so we can play videos from thousands of web sites even live videos and It will be regrettable to let chromium browsers outpace firefox web based Browsers .... because we will have to use chrome and chromium forks to be able to use the program I mentioned
